token/program-2022/src/extension/mod.rs

```diff
@@ -300,8 +329,7 @@ impl<'data, S: BaseState> StateWithExtensionsMut<'data, S> {
     }
 
     /// Packs base state data into the base data portion
-    pub fn pack_base(&mut self, new_base: S) {
-        self.base = new_base;
+    pub fn pack_base(&mut self) {
```

Comment:
I found it strange that `pack_base()` made the caller provider a new base when `unpack()` already populated `self.base`.  The user seems more likely to simply modify `self.base` then copy it elsewhere for modification

Reply:
Makes more sense, definitely!

Reply:
I landed this change independently